test/jdk/javax/swing/JComboBox/DisabledComboBoxFontTestAuto.java

```diff
@@ -90,53 +93,37 @@ private static void paintCombo() {
-        ImageIO.write(disabledImage, "png", new File(testDir
-                + "/" + lafName + "Disabled.png"));
-        ImageIO.write(enabledImage2, "png", new File(testDir
-                + "/" + lafName + "EnabledDLCR.png"));
-        ImageIO.write(disabledImage2, "png", new File(testDir
-                + "/" + lafName + "DisabledDLCR.png"));
-
-        boolean isIdentical = true;
         Color eColor1, eColor2, dColor1, dColor2;
 
         // Use center line to compare RGB values
-        int y = 10;
+        int y = enabledImage.getHeight() / 2;
         for (int x = (enabledImage.getWidth() / 2) - 20;
              x < (enabledImage.getWidth() / 2) + 20; x++) {
-            // Nimbus has a pixel offset in coordinates since Nimbus is 2px
-            // smaller in width than other L&F's
-            if (lafName.equals("Nimbus")) {
-                eColor1 = new Color(enabledImage.getRGB(x + 1, y));
-                eColor2 = new Color(enabledImage2.getRGB(x, y));
-                dColor1 = new Color(disabledImage.getRGB(x + 1, y));
-                dColor2 = new Color(disabledImage2.getRGB(x, y));
-            } else {
-                eColor1 = new Color(enabledImage.getRGB(x, y));
-                eColor2 = new Color(enabledImage2.getRGB(x, y));
-                dColor1 = new Color(disabledImage.getRGB(x, y));
-                dColor2 = new Color(disabledImage2.getRGB(x, y));
-            }
+            eColor1 = new Color(enabledImage.getRGB(x, y));
+            eColor2 = new Color(enabledImage2.getRGB(x, y));
+            dColor1 = new Color(disabledImage.getRGB(x, y));
+            dColor2 = new Color(disabledImage2.getRGB(x, y));
+
             if ((!isColorMatching(eColor1, eColor2)) || (!isColorMatching(dColor1, dColor2))) {
-                isIdentical = false;
-                break;
+                failingLafs.append(lafName + ", ");
+                ImageIO.write(enabledImage, "png", new File(testDir
+                        + "/" + lafName + "Enabled.png"));
+                ImageIO.write(disabledImage, "png", new File(testDir
+                        + "/" + lafName + "Disabled.png"));
+                ImageIO.write(enabledImage2, "png", new File(testDir
+                        + "/" + lafName + "EnabledDLCR.png"));
+                ImageIO.write(disabledImage2, "png", new File(testDir
+                        + "/" + lafName + "DisabledDLCR.png"));
+                return;
             }
         }
-
-        if (isIdentical) {
-            System.out.println("PASSED");
-        } else {
-            failingLafs.append(lafName + ", ");
-        }
+        System.out.println("Test Passed: " + lafName);
     }
 
     private static boolean isColorMatching(Color c1, Color c2) {
         if ((c1.getRed() != c2.getRed())
-                || (c1.getBlue() != c2.getBlue())
-                || (c1.getGreen() != c2.getGreen())) {
-
+            || (c1.getBlue() != c2.getBlue())
```

Comment:
This condition can be optimized since the test is done for all LAF and for each this method is called twice. Instead of using ||, using && would optimized slightly. You can check for `true` where it checks for `==` and returns true. Its just a suggestion.

Reply:
> This condition can be optimized since the test is done for all LAF and for each this method is called twice. Instead of using ||, using && would optimized slightly. You can check for `true` where it checks for `==` and returns true. Its just a suggestion.

I think current comparison with || is ok as if any of the RGB color component doesn't match, test should fail.

Reply:
I meant to use `&&` so that comparisons for failure case fails faster than `||`. 
Similar to this:
` if ((c1.getRed() == c2.getRed())
            && (c1.getBlue() == c2.getBlue())
            && (c1.getGreen() == c2.getGreen())) {
           return true;

} else {

 System.out.println(lafName + " Enabled RGB failure: "
                    + c1.getRed() + ", "
                    + c1.getBlue() + ", "
                    + c1.getGreen() + " vs "
                    + c2.getRed() + ", "
                    + c2.getBlue() + ", "
                    + c2.getGreen());
            return false;
 }`

Reply:
The logical || operator doesn't check second condition if first condition is true. It checks second condition only if first one is false.
So as per current logic, if RED component is not equal, it doesn't check for GREEN and BLUE component and fails.

Reply:
Exactly, same thing applies to && and hence in failure condition || checks for all the colors while && drops off at first failure only. Which is why I suggested it would be better for failure cases.

Reply:
Yeah, its `!=` here right, then it'll fail fast for failure conditions...... Then no need to optimize......

Reply:
In && operator if condition evaluate to false it will not check further.
Similarly, in || operator if condition evaluate to true it will not check further.
So if c1.getRed() != c2.getRed() evaluates to true, it will not check further and fail immediately.